app/src/main/java/com/github/h3lp3rs/h3lp/database/Databases.kt

```diff
@@ -14,5 +25,53 @@ enum class Databases {
+            val skills = skillStorage.getObjectOrDefault(globalContext.getString(R.string.my_skills_key), HelperSkills::class.java, null)
+            if(skills == null) return
+            else {
+                val db = databaseOf(NEW_EMERGENCIES)
+                // Utility function to add a specific listener to a specific key if present
+                fun activateListener(isPresent: Boolean, key: String) {
+                    if(isPresent) {
+                        db.addListenerIfNotPresent(key, Int::class.java) { id ->
+                            // Look if we already encountered this id
+                            val emergencyStorage = storageOf(EMERGENCIES_RECEIVED)
+                            if(emergencyStorage.getBoolOrDefault(id.toString(), false)) {
+                                return@addListenerIfNotPresent
+                            }
+                            // Never see this emergency again later
+                            emergencyStorage.setBoolean(id.toString(), true)
+                            // Send notification only if the associated object still exists
+                            databaseOf(EMERGENCIES).getObject(id.toString(), EmergencyInformation::class.java)
+                                .thenAccept {
+                                    createNotificationChannel(globalContext)
+                                    val intent = Intent(globalContext, HelpPageActivity::class.java)
+                                    // Data to transfer to the help page activity
+                                    val b = Bundle()
+                                    b.putString(EXTRA_EMERGENCY_KEY, it.id)
+                                    b.putStringArrayList(EXTRA_HELP_REQUIRED_PARAMETERS, it.meds)
+                                    b.putDouble(EXTRA_DESTINATION_LAT, it.latitude)
+                                    b.putDouble(EXTRA_DESTINATION_LONG, it.longitude)
+                                    intent.putExtras(b)
+                                    sendIntentNotification(globalContext, globalContext.getString(R.string.emergency),
+                                        globalContext.getString(R.string.need_help), intent)
+                            }
+                        }
+                    }
+                }
+                // Chain of listener instantiations
+                activateListener(skills.hasVentolin, globalContext.getString(R.string.asthma_med))
+                activateListener(skills.isMedicalPro, globalContext.getString(R.string.med_pro))
+                activateListener(skills.hasEpipen, globalContext.getString(R.string.epipen))
+                activateListener(skills.hasInsulin, globalContext.getString(R.string.Insulin))
+                activateListener(skills.hasFirstAidKit, globalContext.getString(R.string.first_aid_kit))
+                activateListener(skills.knowsCPR, globalContext.getString(R.string.cpr))
+            }
+        }

```

Comment:
This a general remark about the help listener.

Doesn't  listener have only the scope of the activity they were launched in? if this is the case this can cause some problem for our notification.

if the  user close the app or relaunch his phone the listener will be dead then and the user will not get any help notification. This could clearly reduce the utility of our app. Nothing to fix this week but if this is the case we need to add this bug to the backlog

Reply:
Well I tested this and even when we close the app (and browse the web for example) it still triggers.
When we relaunch the phone, the listeners will get automatically re-added if needed (while it has not been stored locally).

So maybe in some cases where we close the phone, it may not trigger ? I don't know, since the MainActivity will not get destroyed either way...